client/components/checklist/checklist.js

```diff
@@ -36,9 +37,18 @@ class Checklist extends PureComponent {
 		this.notifyCompletion();
 	}
 
-	componentWillReceiveProps( { siteId } ) {
+	componentWillReceiveProps( { siteId, siteIsUnlaunched } ) {
 		if ( siteId !== this.props.siteId ) {
 			this.setState( { expandedTaskIndex: undefined } );
+			return;
+		}
+		if ( ! siteIsUnlaunched && this.props.siteIsUnlaunched ) {
+			const { expandedTaskIndex } = this.state;
+			if (
+				'site_launched' === get( this, [ 'props', 'children', expandedTaskIndex, 'props', 'id' ] )
+			) {
+				this.setState( { expandedTaskIndex: undefined } );
+			}
```

Comment:
I think it would be better for this logic to be in the `<WpcomChecklistComponent>`. The generic checklist component probably shouldn't have any concept of "unlaunched". I've tested with jetpack though and nothing breaks.